### ISSUE TITLE
Fails to build on Fedora 40 (Hyprland 0.42)

### ISSUE THREAD

**Issue by garrettwp** [dawsers/hyprscroller#50]
I am trying to build hyprscroller on Fedora 40 using Hyprland 0.42 from the copr repo (coprs/solopasha/hyprland) with hyprpm and get the following error:

```
hyprpm -v add https://github.com/dawsers/hyprscroller
[v] version returned: Hyprland, built from branch  at commit 9a09eac79b85c846e3a865a9078a3f8ff65a9259  (props: bump version to 0.42.0).
Date: Wed Aug 7 19:17:10 2024
Tag: v0.42.0, commits: 5069

flags: (if any)


[v] parsed commit 9a09eac79b85c846e3a865a9078a3f8ff65a9259 at branch  on Wed Aug 7 19:17:10 2024, commits 5069
✔ adding a new plugin repository from https://github.com/dawsers/hyprscroller
  MAKE SURE that you trust the authors. DO NOT install random plugins without verifying the code and author.
  Are you sure? [Y/n] y

 → Cloning https://github.com/dawsers/hyprscroller
✔ cloned
✔ found hyprpm manifest
✔ parsed manifest, found 1 plugins:
 → hyprscroller by dawser version
✔ Hyprland headers OK
 → Building hyprscroller
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╍━━━━━━━━━━━━━━━━━━━  3 / 5  Building plugin(s)[v] shell returned:  -> cd /tmp/hyprpm/garrett && PKG_CONFIG_PATH="/home/garrett/.local/share/hyprpm/headersRoot/share/pkgconfig" make release
cmake -B ./Release -DCMAKE_BUILD_TYPE=Release -DCMAKE_PREFIX_PATH=
-- The CXX compiler identification is GNU 14.2.1
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "2.1.1")
-- Checking for module 'hyprland'
--   Found hyprland, version 0.42.0
-- Checking for modules 'hyprutils;hyprlang;pixman-1;libdrm'
--   Found hyprutils, version 0.2.1
--   Found hyprlang, version 0.5.2
--   Found pixman-1, version 0.43.4
--   Found libdrm, version 2.4.123
-- Configuring done (0.6s)
-- Generating done (0.0s)
-- Build files have been written to: /tmp/hyprpm/garrett/Release
cmake --build ./Release -j
gmake[1]: Entering directory '/tmp/hyprpm/garrett/Release'
[ 25%] Building CXX object CMakeFiles/hyprscroller.dir/src/dispatchers.cpp.o
[ 75%] Building CXX object CMakeFiles/hyprscroller.dir/src/scroller.cpp.o
[ 75%] Building CXX object CMakeFiles/hyprscroller.dir/src/main.cpp.o
In file included from /tmp/hyprpm/garrett/src/main.cpp:6:
/tmp/hyprpm/garrett/src/scroller.h:20:43: error: ‘MONITORID’ does not name a type
   20 |     virtual void recalculateMonitor(const MONITORID &monitor_id);
      |                                           ^~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:40:28: error: ‘WORKSPACEID’ has not been declared
   40 |     void cycle_window_size(WORKSPACEID workspace, int step);
      |                            ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:41:21: error: ‘WORKSPACEID’ has not been declared
   41 |     void move_focus(WORKSPACEID workspace, Direction);
      |                     ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:42:22: error: ‘WORKSPACEID’ has not been declared
   42 |     void move_window(WORKSPACEID workspace, Direction);
      |                      ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:43:23: error: ‘WORKSPACEID’ has not been declared
   43 |     void align_window(WORKSPACEID workspace, Direction);
      |                       ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:44:28: error: ‘WORKSPACEID’ has not been declared
   44 |     void admit_window_left(WORKSPACEID workspace);
      |                            ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:45:29: error: ‘WORKSPACEID’ has not been declared
   45 |     void expel_window_right(WORKSPACEID workspace);
      |                             ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:46:23: error: ‘WORKSPACEID’ has not been declared
   46 |     void reset_height(WORKSPACEID workspace);
      |                       ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:47:19: error: ‘WORKSPACEID’ has not been declared
   47 |     void set_mode(WORKSPACEID workspace, Mode);
      |                   ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:48:19: error: ‘WORKSPACEID’ has not been declared
   48 |     void fit_size(WORKSPACEID workspace, FitSize);
      |                   ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:49:26: error: ‘WORKSPACEID’ has not been declared
   49 |     void toggle_overview(WORKSPACEID workspace);
      |                          ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:57:29: error: ‘WORKSPACEID’ has not been declared
   57 |     Row *getRowForWorkspace(WORKSPACEID workspace);
      |                             ^~~~~~~~~~~
In file included from /tmp/hyprpm/garrett/src/dispatchers.cpp:9:
/tmp/hyprpm/garrett/src/scroller.h:20:43: error: ‘MONITORID’ does not name a type
   20 |     virtual void recalculateMonitor(const MONITORID &monitor_id);
      |                                           ^~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:40:28: error: ‘WORKSPACEID’ has not been declared
   40 |     void cycle_window_size(WORKSPACEID workspace, int step);
      |                            ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:41:21: error: ‘WORKSPACEID’ has not been declared
   41 |     void move_focus(WORKSPACEID workspace, Direction);
      |                     ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:42:22: error: ‘WORKSPACEID’ has not been declared
   42 |     void move_window(WORKSPACEID workspace, Direction);
      |                      ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:43:23: error: ‘WORKSPACEID’ has not been declared
   43 |     void align_window(WORKSPACEID workspace, Direction);
      |                       ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:44:28: error: ‘WORKSPACEID’ has not been declared
   44 |     void admit_window_left(WORKSPACEID workspace);
      |                            ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:45:29: error: ‘WORKSPACEID’ has not been declared
   45 |     void expel_window_right(WORKSPACEID workspace);
      |                             ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:46:23: error: ‘WORKSPACEID’ has not been declared
   46 |     void reset_height(WORKSPACEID workspace);
      |                       ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:47:19: error: ‘WORKSPACEID’ has not been declared
   47 |     void set_mode(WORKSPACEID workspace, Mode);
      |                   ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:48:19: error: ‘WORKSPACEID’ has not been declared
   48 |     void fit_size(WORKSPACEID workspace, FitSize);
      |                   ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:49:26: error: ‘WORKSPACEID’ has not been declared
   49 |     void toggle_overview(WORKSPACEID workspace);
      |                          ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:57:29: error: ‘WORKSPACEID’ has not been declared
   57 |     Row *getRowForWorkspace(WORKSPACEID workspace);
      |                             ^~~~~~~~~~~
In file included from /tmp/hyprpm/garrett/src/scroller.cpp:14:
/tmp/hyprpm/garrett/src/scroller.h:20:43: error: ‘MONITORID’ does not name a type
   20 |     virtual void recalculateMonitor(const MONITORID &monitor_id);
      |                                           ^~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:40:28: error: ‘WORKSPACEID’ has not been declared
   40 |     void cycle_window_size(WORKSPACEID workspace, int step);
      |                            ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:41:21: error: ‘WORKSPACEID’ has not been declared
   41 |     void move_focus(WORKSPACEID workspace, Direction);
      |                     ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:42:22: error: ‘WORKSPACEID’ has not been declared
   42 |     void move_window(WORKSPACEID workspace, Direction);
      |                      ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:43:23: error: ‘WORKSPACEID’ has not been declared
   43 |     void align_window(WORKSPACEID workspace, Direction);
      |                       ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:44:28: error: ‘WORKSPACEID’ has not been declared
   44 |     void admit_window_left(WORKSPACEID workspace);
      |                            ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:45:29: error: ‘WORKSPACEID’ has not been declared
   45 |     void expel_window_right(WORKSPACEID workspace);
      |                             ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:46:23: error: ‘WORKSPACEID’ has not been declared
   46 |     void reset_height(WORKSPACEID workspace);
      |                       ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:47:19: error: ‘WORKSPACEID’ has not been declared
   47 |     void set_mode(WORKSPACEID workspace, Mode);
      |                   ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:48:19: error: ‘WORKSPACEID’ has not been declared
   48 |     void fit_size(WORKSPACEID workspace, FitSize);
      |                   ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:49:26: error: ‘WORKSPACEID’ has not been declared
   49 |     void toggle_overview(WORKSPACEID workspace);
      |                          ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.h:57:29: error: ‘WORKSPACEID’ has not been declared
   57 |     Row *getRowForWorkspace(WORKSPACEID workspace);
      |                             ^~~~~~~~~~~
gmake[3]: *** [CMakeFiles/hyprscroller.dir/build.make:76: CMakeFiles/hyprscroller.dir/src/main.cpp.o] Error 1
gmake[3]: *** Waiting for unfinished jobs....
/tmp/hyprpm/garrett/src/scroller.cpp:897:5: error: ‘WORKSPACEID’ does not name a type
  897 |     WORKSPACEID get_workspace() const { return workspace; }
      |     ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:1554:5: error: ‘WORKSPACEID’ does not name a type
 1554 |     WORKSPACEID workspace;
      |     ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp: In constructor ‘Row::Row(PHLWINDOW)’:
/tmp/hyprpm/garrett/src/scroller.cpp:865:11: error: class ‘Row’ does not have any field named ‘workspace’
  865 |         : workspace(window->workspaceID()), reorder(Reorder::Auto),
      |           ^~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp: At global scope:
/tmp/hyprpm/garrett/src/scroller.cpp:1566:6: error: ‘Row* ScrollerLayout::getRowForWorkspace’ is not a static data member of ‘class ScrollerLayout’
 1566 | Row *ScrollerLayout::getRowForWorkspace(WORKSPACEID workspace) {
      |      ^~~~~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:1566:41: error: ‘WORKSPACEID’ was not declared in this scope
 1566 | Row *ScrollerLayout::getRowForWorkspace(WORKSPACEID workspace) {
      |                                         ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp: In member function ‘virtual void ScrollerLayout::onWindowCreatedTiling(PHLWINDOW, eDirection)’:
/tmp/hyprpm/garrett/src/scroller.cpp:1589:5: error: ‘WORKSPACEID’ was not declared in this scope
 1589 |     WORKSPACEID wid = window->workspaceID();
      |     ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:1590:33: error: ‘wid’ was not declared in this scope
 1590 |     auto s = getRowForWorkspace(wid);
      |                                 ^~~
/tmp/hyprpm/garrett/src/scroller.cpp: At global scope:
/tmp/hyprpm/garrett/src/scroller.cpp:1700:47: error: ‘MONITORID’ does not name a type
 1700 | void ScrollerLayout::recalculateMonitor(const MONITORID &monitor_id)
      |                                               ^~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp: In member function ‘virtual PHLWINDOW ScrollerLayout::getNextWindowCandidate(PHLWINDOW)’:
/tmp/hyprpm/garrett/src/scroller.cpp:1914:5: error: ‘WORKSPACEID’ was not declared in this scope
 1914 |     WORKSPACEID workspace_id = g_pCompositor->m_pLastMonitor->activeWorkspaceID();
      |     ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:1915:33: error: ‘workspace_id’ was not declared in this scope; did you mean ‘CWorkspace’?
 1915 |     auto s = getRowForWorkspace(workspace_id);
      |                                 ^~~~~~~~~~~~
      |                                 CWorkspace
/tmp/hyprpm/garrett/src/scroller.cpp: In member function ‘virtual Hyprutils::Math::Vector2D ScrollerLayout::predictSizeForNewWindowTiled()’:
/tmp/hyprpm/garrett/src/scroller.cpp:1956:5: error: ‘WORKSPACEID’ was not declared in this scope
 1956 |     WORKSPACEID workspace_id = g_pCompositor->m_pLastMonitor->activeWorkspaceID();
      |     ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:1957:33: error: ‘workspace_id’ was not declared in this scope; did you mean ‘CWorkspace’?
 1957 |     auto s = getRowForWorkspace(workspace_id);
      |                                 ^~~~~~~~~~~~
      |                                 CWorkspace
/tmp/hyprpm/garrett/src/scroller.cpp: At global scope:
/tmp/hyprpm/garrett/src/scroller.cpp:1967:6: error: variable or field ‘cycle_window_size’ declared void
 1967 | void ScrollerLayout::cycle_window_size(WORKSPACEID workspace, int step)
      |      ^~~~~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:1967:40: error: ‘WORKSPACEID’ was not declared in this scope
 1967 | void ScrollerLayout::cycle_window_size(WORKSPACEID workspace, int step)
      |                                        ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:1967:63: error: expected primary-expression before ‘int’
 1967 | void ScrollerLayout::cycle_window_size(WORKSPACEID workspace, int step)
      |                                                               ^~~
/tmp/hyprpm/garrett/src/scroller.cpp:1991:6: error: variable or field ‘move_focus’ declared void
 1991 | void ScrollerLayout::move_focus(WORKSPACEID workspace, Direction direction)
      |      ^~~~~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:1991:33: error: ‘WORKSPACEID’ was not declared in this scope
 1991 | void ScrollerLayout::move_focus(WORKSPACEID workspace, Direction direction)
      |                                 ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:1991:66: error: expected primary-expression before ‘direction’
 1991 | void ScrollerLayout::move_focus(WORKSPACEID workspace, Direction direction)
      |                                                                  ^~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2028:6: error: variable or field ‘move_window’ declared void
 2028 | void ScrollerLayout::move_window(WORKSPACEID workspace, Direction direction) {
      |      ^~~~~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2028:34: error: ‘WORKSPACEID’ was not declared in this scope
 2028 | void ScrollerLayout::move_window(WORKSPACEID workspace, Direction direction) {
      |                                  ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2028:67: error: expected primary-expression before ‘direction’
 2028 | void ScrollerLayout::move_window(WORKSPACEID workspace, Direction direction) {
      |                                                                   ^~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2038:6: error: variable or field ‘align_window’ declared void
 2038 | void ScrollerLayout::align_window(WORKSPACEID workspace, Direction direction) {
      |      ^~~~~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2038:35: error: ‘WORKSPACEID’ was not declared in this scope
 2038 | void ScrollerLayout::align_window(WORKSPACEID workspace, Direction direction) {
      |                                   ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2038:68: error: expected primary-expression before ‘direction’
 2038 | void ScrollerLayout::align_window(WORKSPACEID workspace, Direction direction) {
      |                                                                    ^~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2047:6: error: variable or field ‘admit_window_left’ declared void
 2047 | void ScrollerLayout::admit_window_left(WORKSPACEID workspace) {
      |      ^~~~~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2047:40: error: ‘WORKSPACEID’ was not declared in this scope
 2047 | void ScrollerLayout::admit_window_left(WORKSPACEID workspace) {
      |                                        ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2055:6: error: variable or field ‘expel_window_right’ declared void
 2055 | void ScrollerLayout::expel_window_right(WORKSPACEID workspace) {
      |      ^~~~~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2055:41: error: ‘WORKSPACEID’ was not declared in this scope
 2055 | void ScrollerLayout::expel_window_right(WORKSPACEID workspace) {
      |                                         ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2063:6: error: variable or field ‘set_mode’ declared void
 2063 | void ScrollerLayout::set_mode(WORKSPACEID workspace, Mode mode) {
      |      ^~~~~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2063:31: error: ‘WORKSPACEID’ was not declared in this scope
 2063 | void ScrollerLayout::set_mode(WORKSPACEID workspace, Mode mode) {
      |                               ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2063:59: error: expected primary-expression before ‘mode’
 2063 | void ScrollerLayout::set_mode(WORKSPACEID workspace, Mode mode) {
      |                                                           ^~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2071:6: error: variable or field ‘fit_size’ declared void
 2071 | void ScrollerLayout::fit_size(WORKSPACEID workspace, FitSize fitsize) {
      |      ^~~~~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2071:31: error: ‘WORKSPACEID’ was not declared in this scope
 2071 | void ScrollerLayout::fit_size(WORKSPACEID workspace, FitSize fitsize) {
      |                               ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2071:62: error: expected primary-expression before ‘fitsize’
 2071 | void ScrollerLayout::fit_size(WORKSPACEID workspace, FitSize fitsize) {
      |                                                              ^~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2079:6: error: variable or field ‘toggle_overview’ declared void
 2079 | void ScrollerLayout::toggle_overview(WORKSPACEID workspace) {
      |      ^~~~~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2079:38: error: ‘WORKSPACEID’ was not declared in this scope
 2079 | void ScrollerLayout::toggle_overview(WORKSPACEID workspace) {
      |                                      ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp:2087:8: error: ‘WORKSPACEID’ does not name a type
 2087 | static WORKSPACEID get_workspace_id() {
      |        ^~~~~~~~~~~
/tmp/hyprpm/garrett/src/scroller.cpp: In member function ‘void ScrollerLayout::marks_add(const std::string&)’:
/tmp/hyprpm/garrett/src/scroller.cpp:2103:38: error: ‘get_workspace_id’ was not declared in this scope
 2103 |     PHLWINDOW w = getRowForWorkspace(get_workspace_id())->get_active_window();
      |                                      ^~~~~~~~~~~~~~~~
gmake[3]: *** [CMakeFiles/hyprscroller.dir/build.make:90: CMakeFiles/hyprscroller.dir/src/dispatchers.cpp.o] Error 1
gmake[3]: *** [CMakeFiles/hyprscroller.dir/build.make:104: CMakeFiles/hyprscroller.dir/src/scroller.cpp.o] Error 1
gmake[2]: *** [CMakeFiles/Makefile2:83: CMakeFiles/hyprscroller.dir/all] Error 2
gmake[1]: *** [Makefile:91: all] Error 2
gmake[1]: Leaving directory '/tmp/hyprpm/garrett/Release'
make: *** [Makefile:13: release] Error 2


✖ Plugin hyprscroller failed to build.
  This likely means that the plugin is either outdated, not yet available for your version, or broken.
  If you are on -git, update first.
  Try re-running with -v to see more verbose output.

✔ all plugins built
✔ installed repository
✔ you can now enable the plugin(s) with hyprpm enable
 ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━  5 / 5  Done!
```

**Comment by obmutescences** [comment]
same in 0.42, archlinux
version returned: Hyprland, built from branch  at commit 9a09eac79b85c846e3a865a9078a3f8ff65a9259  (props: bump version to 0.42.0).
Date: Wed Aug 7 19:17:10 2024
Tag: v0.42.0, commits: 5069

**Comment by dawsers** [comment]
This should be fixed here 884b75ac98116754ec1ecc51c142cfe51c391439

Please, update again.

**Comment by garrettwp** [comment]
Confirm working. Thank you
